icepyx/core/APIformatting.py

```diff
@@ -34,6 +34,9 @@ def _fmt_temporal(start, end, key):
     dictionary with properly formatted temporal parameter for CMR search or subsetting
     """
 
+    if (start is None) or (end is None):
+        return dict()
+
```

Comment:
I puzzled for a few moments over this addition. It's to deal with the fact that for a cycles+tracks search you don't have a temporal parameter. Should we just move "temporal" from "default" to "optional" in our possible keys (`APIformatting._get_possible_keys()`)? We'd still be dictating that the user supply enough info to do a search/order through initialization of the query object, so it just takes it out as a mandatory key then. This all will be moot someday when we're hopefully farming this out to the CMR and/or Harmony APIs, but until then...

Reply:
Great point!  That would work.  I just didn't want to break something :sweat_smile: